### PR TITLE
[extension/encoding] Introduce interfaces for encoding extensions

### PR DESCRIPTION
**Pull request by dmitryax**
**Description:** We should have explicit interfaces for the encoding extensions, which should be used by the receivers/exporters instead of marshallers and unmarshallers

**Link to tracking Issue:** https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/28686

### CODE REVIEW

**Comment by VihasMakwana** [comment]
@dmitryax LGTM! Also, there are some checks/unittest failures in CI.

**Review by VihasMakwana — APPROVED** [review]

**Review by dao-jun — APPROVED** [review]
LGTM

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -91,11 +91,12 @@ extension/asapauthextension/                                            @open-te
 extension/awsproxy/                                                     @open-telemetry/collector-contrib-approvers @Aneurysm9 @mxiamxia
 extension/basicauthextension/                                           @open-telemetry/collector-contrib-approvers @jpkrohling @svrakitin @frzifus
 extension/bearertokenauthextension/                                     @open-telemetry/collector-contrib-approvers @jpkrohling @frzifus
+extension/encoding/                                                     @open-telemetry/collector-contrib-approvers @antoine @dao-jun @dmitryax @MovieStoreGuy @VihasMakwana
```
Comment (atoulme) [inline .github/CODEOWNERS:94]:
```suggestion
extension/encoding/                                                     @open-telemetry/collector-contrib-approvers @atoulme @dao-jun @dmitryax @MovieStoreGuy @VihasMakwana
```

Comment (dmitryax) [reply .github/CODEOWNERS:94]: Sorry :)

Comment (dmitryax) [reply .github/CODEOWNERS]: fixed

**Review by atoulme — APPROVED** [review]

**Pull request merged**